### ISSUE TITLE
Random authentication issue on linux

### ISSUE THREAD

**Issue by lbouriez** [bitwarden/clients#9646]
### Steps To Reproduce

We are using bw to authenticate to bitwarden in our CI, was working perfectly until a few days ago where randomly we get:

> Account does not exist/snapshot/clients/apps/cli/build/bw.js:36644
>                 state.accounts[userId].profile = diskAccount.profile;
>                                                              ^
> 
> TypeError: Cannot read properties of null (reading 'profile')
>     at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:36644:62)
>     at Generator.next (<anonymous>)
>     at /snapshot/clients/apps/cli/build/bw.js:36552:71
>     at new Promise (<anonymous>)
>     at /snapshot/clients/apps/cli/build/bw.js:36548:12
>     at /snapshot/clients/apps/cli/build/bw.js:36639:61
>     at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:37299:44)
>     at Generator.next (<anonymous>)
>     at /snapshot/clients/apps/cli/build/bw.js:36552:71
>     at new Promise (<anonymous>)

We get this error randomly once every 10 runs. The API-key used is always the same.

### Expected Result

We should be connected

### Actual Result

We are not connected

### Operating System

Linux

### Shell

Bash

### Build Version

2024.6

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by jtodddd** [comment]
Hi there,

Would you mind letting me know which linux OS you're using, if you are using the native executable or the snap, or chocolatey and how you are authenticating? Are you doing this: https://bitwarden.com/help/cli/#using-an-api-key ?

Thanks in advance!

**Comment by lbouriez** [comment]
I am running inside a docker container based onto Ubuntu 22.04, using the native app taken from the official link (https://vault.bitwarden.com/download/?app=cli&platform=linux)
For the authentication, I am using the API key
```pwsh
$env:BW_CLIENTID = XXX
$env:BW_CLIENTSECRET = XXX
$env:BW_VAULTNAME = XXX
$env:BW_SUBSCRIPTIONNAME = XXX

$sessionKey = & bw login --apikey --raw
```

As I said, it started randomly, was working since almost 2 years now. We are getting the issue a bit less the last few days, like 1 or twice on 40 run... but when I opened the issue it was way more.

**Comment by lbouriez** [comment]
It just happened again last night
```
/snapshot/clients/apps/cli/build/bw.js:36644
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:36644:62)
    at Generator.next (<anonymous>)
    at /snapshot/clients/apps/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)
    at /snapshot/clients/apps/cli/build/bw.js:36548:12
    at /snapshot/clients/apps/cli/build/bw.js:36639:61
    at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:37299:44)
    at Generator.next (<anonymous>)
    at /snapshot/clients/apps/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)

Node.js v20.11.1
```

What if I provide you my BW username used, it's an enterprise account, could it help ?

**Comment by gempir** [comment]
I have the exact same issue on my mac. (`Sonoma 14.5 (23F79)`)

I'm trying to connect raycast to bitwarden cli. After entering my API credentials and then my master password, this error comes up.

```
/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36644
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36644:62)
    at Generator.next (<anonymous>)
    at /opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)
    at state_service_awaiter (/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36548:12)
    at /opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36639:61
    at StateService.<anonymous> (/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:37299:44)
    at Generator.next (<anonymous>)
    at /opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)

Node.js v20.9.0
```

**Comment by DevOpsMage** [comment]
I'm also seeing this issue running 2024.6.0 on Windows 11.

C:\snapshot\clients\apps\cli\build\bw.js:36644
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (C:\snapshot\clients\apps\cli\build\bw.js:36644:62)
    at Generator.next (<anonymous>)
    at C:\snapshot\clients\apps\cli\build\bw.js:36552:71
    at new Promise (<anonymous>)
    at C:\snapshot\clients\apps\cli\build\bw.js:36548:12
    at C:\snapshot\clients\apps\cli\build\bw.js:36639:61
    at StateService.<anonymous> (C:\snapshot\clients\apps\cli\build\bw.js:37299:44)
    at Generator.next (<anonymous>)
    at C:\snapshot\clients\apps\cli\build\bw.js:36552:71
    at new Promise (<anonymous>)

Node.js v20.11.1

**Comment by MatthiasGrandl** [comment]
I am using the Bitwarden CLI to connect to two different selfhosted vaults and I get this error on only one of the vaults. Used to work fine before. MacOS Sonoma.

**Comment by haykh** [comment]
same issue on Arch
```
<HOME>/.nvm/versions/node/v20.15.0/lib/node_modules/@bitwarden/cli/build/bw.js:36644
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (<HOME>/.nvm/versions/node/v20.15.0/lib/node_modules/@bitwarden/cli/build/bw.js:36644:62)
    at Generator.next (<anonymous>)
    at <HOME>/.nvm/versions/node/v20.15.0/lib/node_modules/@bitwarden/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)
    at state_service_awaiter (<HOME>/.nvm/versions/node/v20.15.0/lib/node_modules/@bitwarden/cli/build/bw.js:36548:12)
    at <HOME>/.nvm/versions/node/v20.15.0/lib/node_modules/@bitwarden/cli/build/bw.js:36639:61
    at StateService.<anonymous> (<HOME>/.nvm/versions/node/v20.15.0/lib/node_modules/@bitwarden/cli/build/bw.js:37299:44)
    at Generator.next (<anonymous>)
    at <HOME>/.nvm/versions/node/v20.15.0/lib/node_modules/@bitwarden/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)

Node.js v20.15.0
```

Tried also node v22.2.0. Also tried installing through `pacman` with the same result.

**Comment by gempir** [comment]
It randomly works for me now. I'm not aware of anything I changed, I don't think I even rebooted. So I have no helpful information why it works now.

**Comment by lbouriez** [comment]
Update, after posting we are having it a lot today, already 5 times in a few run

**Comment by msc5** [comment]
On Debian 12 (Node.js v20.14.0) I get this issue every time I run `bw` without exception so far

**Comment by sirmaxru** [comment]
I have the same problem on macosx, just installed cli via brew
after login I got error: `Account does not exist`

and next run of bw login shows this error:
`(node:66545) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36644
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36644:62)
    at Generator.next (<anonymous>)
    at /opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)
    at state_service_awaiter (/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36548:12)
    at /opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36639:61
    at StateService.<anonymous> (/opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:37299:44)
    at Generator.next (<anonymous>)
    at /opt/homebrew/Cellar/bitwarden-cli/2024.6.0/libexec/lib/node_modules/@bitwarden/cli/build/bw.js:36552:71
    at new Promise (<anonymous>)

Node.js v22.3.0`

**Comment by heaio9v2** [comment]
I am receiving the same error using bw.exe version 2024.6.0, domain joined Windows 24H2, and a Bitwarden enterprise account using SSO. Interestingly, the issue does NOT occur with bw.exe version 2024.6.0, domain joined Windows 24H2, and a Bitwarden family account.

`C:\snapshot\clients\apps\cli\build\bw.js:36644
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (C:\snapshot\clients\apps\cli\build\bw.js:36644:62)
    at Generator.next (<anonymous>)
    at C:\snapshot\clients\apps\cli\build\bw.js:36552:71
    at new Promise (<anonymous>)
    at C:\snapshot\clients\apps\cli\build\bw.js:36548:12
    at C:\snapshot\clients\apps\cli\build\bw.js:36639:61
    at StateService.<anonymous> (C:\snapshot\clients\apps\cli\build\bw.js:37299:44)
    at Generator.next (<anonymous>)
    at C:\snapshot\clients\apps\cli\build\bw.js:36552:71
    at new Promise (<anonymous>)

Node.js v20.11.1`

**Comment by sirmaxru** [comment]
found some workaround:
If I exec like this- no problems occure. But I want it to work in default way. 
`BITWARDENCLI_APPDATA_DIR=~/.bw/ bw generate`

**Comment by Viktonaft** [comment]
@sirmaxru export BITWARDENCLI_APPDATA_DIR=~/.bw/ helps, thanks!

**Comment by MatthiasGrandl** [comment]
Still facing this issue. I already do specificy the `BITWARDENCLI_APPDATA_DIR` and that doesn't change anything.

**Comment by Daankrol** [comment]
Is there any update on this? I am still experiencing this problem on Mac OS Sonoma

**Comment by martingraf4** [comment]
**TL;DR** workaround that worked for us:

1. `npm i -g @bitwarden/cli@2024.6.0`
2. `BITWARDENCLI_APPDATA_DIR=<some path that you have not used with a newer version of the bitwarden cli yet> bw ...`


The problem seems to be related to the latest version(s) of the CLI. Version 2024.6.0 works fine for us. Note that the newer version(s) seems to corrupt the appdata directory. Once corrupted, also older bitwarden versions will fail when the same appdata directory is used.

**Comment by eviatarh** [comment]
I am still experiencing this issue using a node pod with @bitwarden/cli@2024.7.2 - the workaround of BITWARDENCLI_APPDATA_DIR didn't fix the issue which keep happening in my production service from time to time. 
In addition, I couldn't find a CLI release with this merged commit, even with the latest version published ([v2024.8.2](https://github.com/bitwarden/clients/releases/tag/cli-v2024.8.2)) : https://github.com/bitwarden/clients/commit/11eba8d77902c8b115da541cc09a0472cf74aca3

I am not so familiar with the way you manage the releases of this repo and not sure if it's the right place to ask, can you please assist? 🙏

**Comment by maxwassiljew** [comment]
I’m experiencing the same issue, which occurs intermittently. Initially, I encountered the error while using the latest version. After reading through the comments, I downgraded to 2024.6.0, but the problem persists. For context, I’m running the bw CLI inside a Docker container.

**Comment by Lozbik** [comment]
I have a similar problem, I installed it according to the instructions from the [official website](https://bitwarden.com/help/cli/). And immediately after installation, I started getting errors when trying to execute any bw command. An attempt to set `BITWARDENCLI_APPDATA_DIR` gives a result, but only for one session, then you need to create a new directory again. At the same time, if you create a new directory, replace the value of `BITWARDENCLI_APPDATA_DIR` with the bw command, then return the value of the variable `BITWARDENCLI_APPDATA_DIR` back, then the bw commands will continue to be executed. 
I'm trying to use Royal TSX to connect to Bitwarden, but I keep getting an error.

My Error:
```
An error occurred while executing the dynamic folder's script: /snapshot/clients/apps/cli/build/bw.js:39426
state.accounts[userId].profile = diskAccount.profile;
^

TypeError: Cannot read properties of null (reading 'profile')
at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:39426:62)
at Generator.next (<anonymous>)
at /snapshot/clients/apps/cli/build/bw.js:39337:71
at new Promise (<anonymous>)
at /snapshot/clients/apps/cli/build/bw.js:39333:12
at /snapshot/clients/apps/cli/build/bw.js:39421:61
at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:39898:44)
at Generator.next (<anonymous>)
at /snapshot/clients/apps/cli/build/bw.js:39337:71
at new Promise (<anonymous>)

Node.js v20.11.1
/snapshot/clients/apps/cli/build/bw.js:39426
state.accounts[userId].profile = diskAccount.profile;
^

TypeError: Cannot read properties of null (reading 'profile')
at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:39426:62)
at Generator.next (<anonymous>)
at /snapshot/clients/apps/cli/build/bw.js:39337:71
at new Promise (<anonymous>)
at /snapshot/clients/apps/cli/build/bw.js:39333:12
at /snapshot/clients/apps/cli/build/bw.js:39421:61
at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:39898:44)
at Generator.next (<anonymous>)
at /snapshot/clients/apps/cli/build/bw.js:39337:71
at new Promise (<anonymous>)

Node.js v20.11.1
```

My configuration:
- macOS 14.7 23H124
- Node.js v20.11.1
- bw version 2024.8.2

**Comment by Brennwert** [comment]
We experience the same with 2024.12.0 on Debian bullseye (Python Docker image) against a self-hosted Vaultwarden server. Sporadically bw login results in "Account does not exist".
We currently work around the issue using a simple loop, which is solving it for now:

```
max_retry=5
counter=0
until BW_SESSION=$(bw login --raw $BITWARDEN_ACCOUNT $BITWARDEN_PW)
do
  [[ $counter -eq $max_retry ]] && echo "Failed!" && exit 1
  echo "bw login failed, trying again -- $counter / $max_retry"
  counter=$((counter+1))
  sleep 1
  bw logout || true
done
export BW_SESSION
```
